go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -7401,9 +7401,24 @@ var DoltCommitTests = []queries.ScriptTest{
 				Expected: []sql.Row{{sql.NewOkResult(1)}},
 			},
 			{
-				Query:            "CALL DOLT_COMMIT('-ALL', '-m', 'update table t');",
+				Query:            "CALL DOLT_COMMIT('-ALL', '-m', 'update table terminator');",
 				SkipResultsCheck: true,
 			},
+			// check initial commit
+			{
+				Query:    "select message from dolt_log limit 1",
+				Expected: []sql.Row{{"update table terminator"}},
+			},
+			// amend the commit
+			{
+				Query:            "CALL DOLT_COMMIT('-amend', '-m', 'update table t');",
+				SkipResultsCheck: true,
+			},
+			// check amended commit
+			{
```

Comment:
Good idea, will add a few more commit/amend combinations.

Regarding the `dolt_status`, `dolt_log`, and `dolt_diff_<tablename> tables: I've been able to run the same steps through CLI and SQL (`dolt commit --amend ...` vs `call dolt_commit('--amend'...)`) and I can see that both are behaving identically. But that's just some manual testing. Should I try to add tests that verify the data in these various tables?

Reply:
Thank you! Yeah... my thinking is that since `--amend` allows you to change several things (e.g. commit message, adding included files, removing included files) we should test a little deeper to make sure it's working as expected. Some of those system tables are likely helpful to writing those tests to verify, but let me know if you get stuck and can't find the data you need.

Reply:
Added more tests, including updating the commit message, adding an extra change to a commit, and updating a message in a merge commit. Let me know if these new tests were added correctly/in the right spot.

I'm not sure how to test "removing included files", can you share what that operation might look like?

Reply:
Discussed offline, latest update has the updated tests.